backend/src/main/kotlin/org/icpclive/Config.kt

```diff
@@ -1,9 +1,39 @@
 package org.icpclive
 
+import io.ktor.server.application.*
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+import java.io.File
 import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.exists
 
 object Config {
     lateinit var configDirectory: Path
+    lateinit var presetsDirectory: Path
+    lateinit var mediaDirectory: Path
     lateinit var creds: Map<String, String>
     var allowUnsecureConnections: Boolean = false
-}
\ No newline at end of file
+
+    fun initialize(environment: ApplicationEnvironment) {
```

Comment:
To be honest, loocks a bit overloaded now. Maybe just have 

lateinit var Config and normal constructor?